### ISSUE TITLE
va-text-input and va-textarea error message not announced by screen readers

### ISSUE THREAD

**Issue by rmessina1010** [department-of-veterans-affairs/vets-design-system-documentation#2572]
# Bug Report

- [x] I’ve searched for any related issues and avoided creating a duplicate issue.

## What happened
We are encountering some issues with screen readers not announcing errors on MHV Secure Messaging forms when composing a message.  Specifically, this is happening with va-text-input and va-textarea.  Error messages aren't announced when tabbing to them, nor when we manually focus on the corresponding error.  We are wondering if anyone else has experienced this and what they've done to address it?  It seems to be affecting multiple forms.

## What I expected to happen

- The text of the error to be announced,  while focus is maintained on input  field ( note: the later is happening, as. the error  message text itself is not focusable)

## Reproducing
- Formation version:
- Device: Macbook
- Browser:  Chrome 122.0.6261

Steps to reproduce:

1.
2.
3.
4.

## Urgency
How urgent is this request? Please select the approriate option below and/or provide details
- [ ] This bug is blocking work currently in progress
- [ ] This bug is affecting work currently in progress but we have a workaround
- [ ] This bug is blocking work planned within the next few sprints
- [ ] This bug is not blocking any work
- [ ] Other

**Comment by caw310** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Platform-Design-System-5f8de67192551b0012ebb802/issues/department-of-veterans-affairs/vets-design-system-documentation/2572?planning-poker) with Zenhub @Andrew565 @ataker @harshil1793 @it-harrison @jamigibbs @micahchiang @nickjg231 @powellkerry @rmessina1010 @rsmithadhoc

**Comment by rsmithadhoc** [comment]
The error messages must be associated to their respective controls by placing that error within the `label` or using `aria-describedby`.
- I believe we show the error and give them `role="alert"` when the error message is triggered, which will announce it when it happens. However, if they tab away and return to the field or if an edit form contains an error on page load, it won't tell a screen reader user anything when tabbing to it.

**Comment by rmessina1010** [comment]
@rsmithadhoc 
I found this article: https://stackoverflow.com/questions/75178816/nvda-chrome-reads-anchor-text-inside-a-hidden-div-twice-when-displayed/75189223#75189223
The gist being  that if an element, with role,  switches display context it is not read by Voice Over, suggesting wrapping said element inside another element with `aria-live`.

If I am looking at the right place, perhaps applying `aria-live="polite"` at `#input-error-message`
```
            <span id="input-error-message" role="alert">
              {error && (
                <Fragment>
                  <span class="usa-sr-only">{i18next.t('error')}</span>
                  <span class="usa-error-message">{error}</span>
                </Fragment>
              )}
            </span>
```
or at its container element (  `<div class="input-wrap">...</div>`  or `Host`?)

What do you think?

**Comment by rsmithadhoc** [comment]
> @rsmithadhoc I found this article: https://stackoverflow.com/questions/75178816/nvda-chrome-reads-anchor-text-inside-a-hidden-div-twice-when-displayed/75189223#75189223 The gist being that if an element, with role, switches display context it is not read by Voice Over, suggesting wrapping said element inside another element with `aria-live`.
> 
> If I am looking at the right place, perhaps applying `aria-live="polite"` to `#input-error-message`
> 
> ```
>             <span id="input-error-message" role="alert">
>               {error && (
>                 <Fragment>
>                   <span class="usa-sr-only">{i18next.t('error')}</span>
>                   <span class="usa-error-message">{error}</span>
>                 </Fragment>
>               )}
>             </span>
> ```
> 
> or at its container element ( `<div class="input-wrap">...</div>` or `Host`?)
> 
> What do you think?

I think the issue is more to do with tabbing to the field, rather than alerting on error or alerting to changing content, which is what `aria-live` is used for. But I'll test that as well to make sure it's working and reference this info if needed, thanks!

**Comment by rsmithadhoc** [comment]
In troubleshooting this, I tried a basic example of `aria-describedby` that _should_ read out the description when tabbing to the field, but it did not. So this is not an issue with our component. I searched for Mac and VoiceOver changes and found this [thread](https://bugs.webkit.org/show_bug.cgi?id=262895). VoiceOver is no longer reading out `aria-describedby` by default, but does pick it up and display it under another menu.

It currently only reads `aria-describedby` with VoiceOver and Firefox. It did not work on VoiceOver with Safari or Chrome.
